roles/workstation/tasks/main.yml

```diff
@@ -0,0 +1,278 @@
+- name: Configure firewall
+  copy:
+    src: workstation.rules
+    dest: /etc/iptables.d/workstation.rules
+    owner: root
+    group: root
+    mode: 0640
+  notify:
+  - iptables
+
+- name: Install graphics libraries
+  xbps:
+    pkg:
+      - libXrandr-32bit
+      - libXt-32bit
+      - mesa-ati-dri-32bit
+      - mesa-intel-dri-32bit
+    state: present
+
+- name: Install various system services
+  xbps:
+    pkg:
+      - ConsoleKit2
+      - cgmanager
+    state: present
+
+- name: Install editing suites
+  xbps:
+    pkg:
+      - gimp
+      - inkscape
+      - libreoffice
+    state: present
+
+- name: Install programming tools
+  xbps:
+    pkg:
+      - base-devel
+      - eclipse
+      - codeblocks
+      - gdb
+      - ghc
+      - git-all
+      - openjdk
+      - R
+      - racket
+      - spim
+      - rust
+      - cargo
+      - texlive-bin
+      - valgrind
+      - xterm
+    state: present
+
+- name: Install additional editors
+  xbps:
+    pkg:
+      - emacs
+      - gedit
+      - neovim
+      - python-neovim
+      - python3.4-neovim
+      - gvim
+      - sublime-text3
+    state: present
+
+- name: Install media playback programs
+  xbps: 
+    pkg:
+      - audacity
+      - alsa-lib
+      - alsa-utils
+      - libdvdcss
+      - mpv
+      - pulseaudio
+      - spotify
+      - vlc
+    state: present
+
+- name: Install Web Browsers
+  xbps: 
+    pkg:
+      - chromium
+      - firefox
+      - google-chrome
+      - opera
+    state: present
+
+- name: Install Scanner programs
+  xbps: 
+    pkg:
+      - simple-scan
+      - sane
+      - sane-epkowa
+      - iscan
+      - iscan-data
+      - iscan-plugin-gt-s650
+    state: present
+    
+- name: Install additional misc. programs
+  xbps: 
+    pkg:
+      - evince
```

Comment:
We can take the variable approach if it can be made to work, but I'd still like to have the separate lists for clarity as we add and remove software.